src/ngraph/pass/algebraic_simplification.cpp

```diff
@@ -726,6 +726,7 @@ static unordered_map<NodeTypeInfo, function<bool(shared_ptr<Node>)>> initialize_
          {op::v0::Multiply::type_info, simplify_multiply},
          {opset3::Gather::type_info, simplify_gather},
          {op::v0::Concat::type_info, simplify_concat},
+         {opset2::ShapeOf::type_info, simplify_gather_shapeof},
```

Comment:
if we have the same optimization needed for particular Op belonging to different `Opsets` Isn't it , we end up implementing and registering same pass to make it work for different Opsets ?